### PR TITLE
Add pyspark-flame (#6998)

### PR DESCRIPTION
**Pull request by benjamb**
This is my first attempt at creating a new package, so I thought I'd pick up one from the backlog of requests. While this seems to tick all the boxes I'm a little uncertain as to whether I need to list openjdk as a runtime requirement, since I don't believe pyspark is of much use without the appropriate jar files.

Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by nowster** [comment]
I am willing to be listed as maintainer.

**Comment by nehaljwani** [comment]
@conda-forge-admin, please restart ci

File: recipes/pyspark-flame/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 807099c77fbaefaeb71217995bd22d203c697c6ce32c8e75ff53e1b1b56d9322
+
+build:
+  number: 0
```
Comment (nehaljwani) [inline recipes/pyspark-flame/meta.yaml:13]: Can this be `noarch: python` ?

xref: https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python

File: recipes/pyspark-flame/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - pypandoc
```
Comment (marcelotrevisani) [inline recipes/pyspark-flame/meta.yaml:18]: I don't think it should be here
```suggestion
```

File: recipes/pyspark-flame/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  host:
+    - pip
+    - pyspark
```
Comment (marcelotrevisani) [inline recipes/pyspark-flame/meta.yaml:21]: I think this package needs ``pyspark`` just to run the project, right?
```suggestion
```

File: recipes/pyspark-flame/meta.yaml (hunk 2 above)
Comment (benjamb) [reply recipes/pyspark-flame/meta.yaml:18]: I must admit I wasn't that clear after reading the documentation surrounding the requirements section. pypandoc is listed in the pyspark-flame's `setup.py` as `setup_requires=['pypandoc'],`, which I thought would have made it appropriate to add to this section?

Comment (marcelotrevisani) [reply recipes/pyspark-flame/meta.yaml:18]: It should be in ``host`` then, not in ``build``
If you want to, next time you can give a go on [Grayskull](https://github.com/marcelotrevisani/grayskull) top generate recipes of packages on ``pypi``

File: recipes/pyspark-flame/meta.yaml (hunk 3 above)
Comment (benjamb) [reply recipes/pyspark-flame/meta.yaml:21]: I'd have thought so yes, this came about as part of running the `conda skeleton` CLI.

Comment (marcelotrevisani) [reply recipes/pyspark-flame/meta.yaml:21]: ``conda-build skeleton`` is not the best tool to generate clean recipes for ``conda``. [Grayskull](https://github.com/marcelotrevisani/grayskull) was started to generate better recipes for conda and eventually replace ``conda-build skeleton``

https://conda-forge.org/blog//2020/03/05/grayskull

File: recipes/pyspark-flame/meta.yaml (hunk 2 above)
Comment (benjamb) [reply recipes/pyspark-flame/meta.yaml:18]: Yeah I just had another read through, makes a bit more sense. I'll take a look at Grayskull.

File: recipes/pyspark-flame/meta.yaml (hunk 1 above)
Comment (benjamb) [reply recipes/pyspark-flame/meta.yaml:13]: Done.

File: recipes/pyspark-flame/meta.yaml (hunk 2 above)
Comment (benjamb) [reply recipes/pyspark-flame/meta.yaml]: Done.

File: recipes/pyspark-flame/meta.yaml (hunk 3 above)
Comment (benjamb) [reply recipes/pyspark-flame/meta.yaml]: Done.

Comment (benjamb) [reply recipes/pyspark-flame/meta.yaml]: When I remove this, conda build no longer respects that pyspark requires python <3.8 and so the CI fails.

Comment (benjamb) [reply recipes/pyspark-flame/meta.yaml]: Sorry I misspoke, this is a result of adding `noarch: python`.

File: recipes/pyspark-flame/meta.yaml (hunk 1 above)
Comment (benjamb) [reply recipes/pyspark-flame/meta.yaml:13]: @nehaljwani I don't believe this is possible, since pyspark requires python <3.8.

**Comment by benjamb** [comment]
I've addressed the above review feedback, what's next for this PR?

@conda-forge-admin, please ping team

**Pull request merged**